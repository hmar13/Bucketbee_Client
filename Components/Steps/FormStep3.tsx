import React, { Dispatch, SetStateAction } from 'react';
import { View, Text, StyleSheet, TextInput } from 'react-native';
import * as Animatable from 'react-native-animatable';
import { Button } from 'react-native-paper';
import theme from '../../styles/theme.style';

type SetCountType = Dispatch<SetStateAction<number>>
type SetEmojiType = Dispatch<SetStateAction<string | undefined>>


interface Props {
  setEmojis: SetEmojiType;
  emojis: string | undefined;
  handleSubmit(): void;
}

const FormStep3: React.FC<Props> = ({ emojis, setEmojis, handleSubmit }) => {
  return (
    <View style={styles.container}>
      <Animatable.View animation="fadeInUpBig" style={styles.wrapper}>
        <Text style={styles.text}>Describe yourself in 3 emojis:</Text>
        <TextInput
          value={emojis}
          onChangeText={(val) => setEmojis(val)}
          placeholder="Emojis go here"
          style={styles.input}
          placeholderTextColor={theme.PRIMARY_COLOR_XLITE}
        />
        <Button
          labelStyle={{ color: theme.PRIMARY_COLOR }}
          style={styles.button}
          onPress={handleSubmit}
        >
          SUBMIT
        </Button>
      </Animatable.View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: theme.PRIMARY_COLOR,
  },
  wrapper: {
    height: 200,
    width: 300,
    justifyContent: 'center',
  },
  input: {
    width: 280,
    height: 50,
    borderBottomColor: theme.PRIMARY_COLOR_XLITE,
    borderBottomWidth: 2,
    color: theme.PRIMARY_COLOR_XLITE,
    fontSize: 22,
    fontFamily: 'Lato_300Light',
  },
  button: {
    backgroundColor: theme.PRIMARY_COLOR_XLITE,
    marginVertical: 20,
    color: theme.PRIMARY_COLOR,
    height: 50,
    width: 100,
    justifyContent: 'center',
  },
  text: {
    color: theme.PRIMARY_COLOR_XLITE,
    fontSize: 24,
    fontFamily: 'Poppins_400Regular',
    paddingBottom: 8,
  },
});

export default FormStep3;
