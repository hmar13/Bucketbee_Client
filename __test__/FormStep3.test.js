import React from 'react';
import { TextInput } from 'react-native';
import { Button } from 'react-native-paper';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import FormStep3 from '../Components/Steps/FormStep3';

describe('<FormStep3 />', () => {
  const setEmojis = jest.fn();
  const handleSubmit = jest.fn();
  const wrapper = shallow(<FormStep3
    setEmojis={setEmojis}
    handleSubmit={handleSubmit}
  />);

  test('Renders correctly', () => {
    const tree = renderer.create(<FormStep3 />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Should render the Text Element', () => {
    expect(wrapper.find('Text').at(0).props().children).toEqual('Describe yourself in 3 emojis:')
  });

  test('Should call setState on TextInput with value entered', () => {
    wrapper.find(TextInput).simulate('changeText', 'new emoji here');
    expect(setEmojis).toHaveBeenCalledWith('new emoji here');
  })

  test('Should call handle function on Button Press', () => {
    wrapper.find(Button).simulate('press');
    expect(handleSubmit).toHaveBeenCalledTimes(1);
  })
});
