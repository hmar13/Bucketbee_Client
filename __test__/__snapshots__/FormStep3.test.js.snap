// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormStep3 /> Renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#0d525f",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    forwardedRef={[Function]}
    style={
      Object {
        "height": 200,
        "justifyContent": "center",
        "opacity": 0,
        "transform": Array [
          Object {
            "translateY": 500,
          },
        ],
        "width": 300,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#9affdc",
          "fontFamily": "Poppins_400Regular",
          "fontSize": 24,
          "paddingBottom": 8,
        }
      }
    >
      Describe yourself in 3 emojis:
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Emojis go here"
      placeholderTextColor="#9affdc"
      rejectResponderTermination={true}
      style={
        Object {
          "borderBottomColor": "#9affdc",
          "borderBottomWidth": 2,
          "color": "#9affdc",
          "fontFamily": "Lato_300Light",
          "fontSize": 22,
          "height": 50,
          "width": 280,
        }
      }
      underlineColorAndroid="transparent"
    />
    <View
      style={
        Object {
          "backgroundColor": "#9affdc",
          "borderColor": "transparent",
          "borderRadius": 4,
          "borderStyle": "solid",
          "borderWidth": 0,
          "color": "#0d525f",
          "elevation": 0,
          "height": 50,
          "justifyContent": "center",
          "marginVertical": 20,
          "minWidth": 64,
          "width": 100,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": undefined,
          }
        }
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 4,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "letterSpacing": 1,
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  undefined,
                  Object {
                    "textTransform": "uppercase",
                  },
                  Object {
                    "color": "#6200ee",
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "color": "#0d525f",
                  },
                ],
              ]
            }
          >
            SUBMIT
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
